shared/libblight_protocol/libblight_protocol.cpp

```
@@ -0,0 +1,665 @@
+    int blight_detach_connection_thread(blight_thread_t* thread){
+        if(thread == nullptr){
+            errno = EINVAL;
+            return -errno;
+        }
+        thread->handle.detach();
+        return 0;
+    }
+    int blight_connection_thread_deref(blight_thread_t* thread){
+        if(thread == nullptr){
+            return 0;
+        }
```

Cppcheck (2.10-2)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)